presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -651,11 +652,22 @@ public void writeJsonValue(JsonGenerator jsonGenerator, Block block, int positio
             }
             else {
                 Block rowBlock = type.getObject(block, position);
-                jsonGenerator.writeStartArray();
-                for (int i = 0; i < rowBlock.getPositionCount(); i++) {
-                    fieldWriters.get(i).writeJsonValue(jsonGenerator, rowBlock, i, properties);
+                if (!properties.isFieldNamesInJsonCastEnabled()) {
+                    jsonGenerator.writeStartArray();
+                    for (int i = 0; i < rowBlock.getPositionCount(); i++) {
+                        fieldWriters.get(i).writeJsonValue(jsonGenerator, rowBlock, i, properties);
+                    }
+                    jsonGenerator.writeEndArray();
+                }
+                else {
+                    List<TypeSignatureParameter> typeSignatureParameters = type.getTypeSignature().getParameters();
+                    jsonGenerator.writeStartObject();
+                    for (int i = 0; i < rowBlock.getPositionCount(); i++) {
+                        jsonGenerator.writeFieldName(typeSignatureParameters.get(i).getNamedTypeSignature().getName().orElse(""));
```

Comment:
will the default value "" cause trouble such as duplicated field name?

Reply:
Nope, we already have a  test for that in https://github.com/prestodb/presto/pull/19513/files#diff-9cb149b592e20c436e946cf96f593ae4f1fbf948513df713f71bdcd772279324R154

Reply:
Great, then I have no concern